pkg/scheduler/objects/application.go

```diff
@@ -1712,6 +1715,17 @@ func (sa *Application) ReplaceAllocation(uuid string) *Allocation {
 	if sa.placeholderData != nil {
 		sa.placeholderData[ph.GetTaskGroup()].Replaced++
 	}
+
+	// check if we have active placeholder allocation left
+	hasReplaceablePlaceholder := false
+	for _, phData := range sa.placeholderData {
+		if phData.Count > (phData.Replaced + phData.TimedOut) {
+			hasReplaceablePlaceholder = true
+			break
+		}
+	}
+	sa.hasPlaceholderAlloc.Store(hasReplaceablePlaceholder)
```

Comment:
This part is crucial and needs to be checked carefully. If we replaced all PHs, we set it back to false so we won't bother with further PH allocations.

Reply:
See earlier comment, placeholder allocation based is more stable and does not need any new looping as we already have the checks for other things